### PR TITLE
fix: prevent slug collision on draft

### LINKED ISSUE #2744: [bug] Saving an item as draft with a slug that is in use causes URL collision
**Describe the bug**
When you save a draft howto, you are allowed to use a slug that is currently used in another howto. This causes an issue when you navigate to your draft howto, as it shares the same URL as an existing howto.

**To Reproduce**
Steps to reproduce the behaviour:

1. Find an existing how to slug
2. Create a new howto with the same slug
3. Save as draft
4. Click on your draft how to
5. You will be navigated to the existing how to, not your own

**Expected behaviour**
Should not be able to save a draft howto with the same slug as an existing howto. 

**Additional context**
Use the same validation requirements for drafts and publishing for the title field

### PR DESCRIPTION
**Pull request by goratt12**
PR Checklist

- [ ] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

Changed the validator function of the title to ignore draft mode.
even if saved as draft, all relevant validation must pass (including unique title/slug)

Changed the validation function of unique title/slug to compare slug against other HowTo `slug` field instead of `previousSlugs` field

## Git Issues

Closes #2744 

## Screenshots/Videos

When saving as draft:
image

When publishing:
image

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Comment by goratt12** [comment]
This notice is now showing when a user is using a title that is already in use (Checked onBlur)
image

File: src/pages/Howto/Content/Common/HowtoFieldTitle.tsx [hunk 1]
```diff
@@ -46,15 +44,25 @@ export const HowtoFieldTitle = (props: IProps) => {
...
         placeholder={placeholder}
-        showCharacterCount
+        showCharacterCount={!duplicateTitleNotice}
+        customOnBlur={handleBlur}
       />
+      {duplicateTitleNotice && (
+        <Card sx={{ bg: 'softblue', padding: 2, marginTop: 1, border: 0 }}>
+          <Text sx={{ fontSize: 1 }}>
+            Did you know there is an existing how-to with the title Existing
+            how-to Title? Using a unique title helps readers decide which how-to
```
Comment (thisislawatts) [inline src/pages/Howto/Content/Common/HowtoFieldTitle.tsx:61]: **questionblocking):** I think that `Existing how-to Title` should be dynamic here, so it loads the value from the form.

Comment (goratt12) [reply src/pages/Howto/Content/Common/HowtoFieldTitle.tsx:61]: How would you recommend implementing it in a dynamic way?
I tried looking in the code for an example but could not find any.
Should I separate the notice card to a different component?

Comment (goratt12) [reply src/pages/Howto/Content/Common/HowtoFieldTitle.tsx:61]: How do you recommend I implement it? 
I couldn't find an example of such notice in the code.

Comment (goratt12) [reply src/pages/Howto/Content/Common/HowtoFieldTitle.tsx]: You are correct I didn't notice it. now is dynamic

**Comment by goratt12** [comment]
I'm getting a strange bug when testing my changes. 
in src\stores\Howto\howto.store.tsx in the class HowtoStore I'm calling the function this.isTitleThatReusesSlug that is implemented in the parent class ModuleStore.
When running the code everything works. but when I run tests, the code doesn't recognize the function this.isTitleThatReusesSlug, it looks like it doesn't recognize that the class HowtoStore is extending ModuleStore.
The error rise in the file src\stores\Howto\howto.store.test.ts
Can someone help me understand why this is happening and how I can resolve it?

**Comment by thisislawatts** [comment]
Sorry @goratt12, I missed your comment here. I haven't been able to recreate the behaviour and the CI seems happy.

**Review by thisislawatts — APPROVED** [review]

**Pull request merged**